cli/command/swarm/init.go

```diff
@@ -44,21 +47,84 @@ func newInitCommand(dockerCli command.Cli) *cobra.Command {
 	flags.BoolVar(&opts.forceNewCluster, "force-new-cluster", false, "Force create a new cluster from current state")
 	flags.BoolVar(&opts.autolock, flagAutolock, false, "Enable manager autolocking (requiring an unlock key to start a stopped manager)")
 	flags.StringVar(&opts.availability, flagAvailability, "active", `Availability of the node ("active"|"pause"|"drain")`)
+	flags.StringVar(&opts.defaultAddrPool, flagDefaultAddrPool, "", "List of default subnet addresses followed by subnet size (format: <subnet,subnet,..:subnet-size)")
+	flags.SetAnnotation(flagDefaultAddrPool, "version", []string{"1.39"})
 	addSwarmFlags(flags, &opts.swarmOptions)
 	return cmd
 }
 
+// getDefaultAddrPool extracts info from address pool string and fills PoolsOpt struct
+func getDefaultAddrPool(defaultAddrPool string) ([]*net.IPNet, int, error) {
+	var size int
+	var err error
+	var defaultPoolList []*net.IPNet
+	if defaultAddrPool == "" {
+		// defaultAddrPool is not defined
+		return nil, size, nil
+	}
+
+	result := strings.Split(defaultAddrPool, ":")
+	if len(result) > 2 {
+		return nil, size, fmt.Errorf("default address pool must be a non-zero IP address or addresses format CIDR,CIDR,...:SUBNET-SIZE")
+	}
+	// if size is not specified default size is 24
+	if len(result) == 1 {
+		size = 24
+	} else {
+		// trim leading and trailing white spaces
+		result[1] = strings.TrimSpace(result[1])
+
+		// get the size from the slice
+		size, err = strconv.Atoi(result[1])
+
+		if err != nil {
+			return nil, size, fmt.Errorf("error in DefaultAddressPool subnet size %s", defaultAddrPool)
+		}
+	}
+
+	// get subnet list
+	subnetlist := strings.Split(result[0], ",")
+
+	for i := range subnetlist {
+		// trim leading and trailing white spaces
+		subnetlist[i] = strings.TrimSpace(subnetlist[i])
+		_, b, err := net.ParseCIDR(subnetlist[i])
+		if err != nil {
+			return nil, size, fmt.Errorf("invalid base pool %q: %v", subnetlist[i], err)
+		}
+		ones, _ := b.Mask.Size()
+		if size <= 0 || size < ones {
+			return nil, size, fmt.Errorf("invalid pool size: %d", size)
+		}
```

Comment:
Could then change the error to:  "subnet size too small for pool" or something like that.

Reply:
done